src/views/disciplinas/hooks/useFilterSubject.js

```diff
@@ -0,0 +1,51 @@
+import { useContext, useState } from 'react'
+import { subjectContext } from './useSubject'
+
+export function useFilterSubject() {
+  const [filteredData, setFilteredData] = useState([])
```

Comment:
Não acho que seja necessário criar state para o `filteredData`. Ele pode ser uma variável normal, e aí você preenche sempre que o `filterBy` for selecionado.

Algo assim:

`const filteredData = filterBy.length ? chama_a_funcao_de_pesquisa : [] `